sys/net/sock/sock_util.c

```diff
@@ -43,11 +43,13 @@ int sock_tl_ep_fmt(const struct _sock_tl_ep *endpoint,
     *addr_str = '\0';
 
     switch (endpoint->family) {
+#if defined(SOCK_HAS_IPV4)
         case AF_INET:
             {
                 addr_ptr = (void*)&endpoint->addr.ipv4;
                 break;
             }
+#endif
 #if defined(SOCK_HAS_IPV6)
         case AF_INET6:
             {
```

Comment:
Wouldn't a simple

```C
    if (IS_ACTIVE(SOCK_HAS_IPV4) && (endpoint->family == AF_INET)) {
        addr_ptr = &endpoint->addr.ipv4;
    }
    else if (IS_ACTIVE(SOCK_HAS_IPV6) && (endpoint->family == AF_INET6)) {
        addr_ptr = &endpoint->addr.ipv6;
    }
    else {
        return -ENOTSUP;
    }
```

or maybe even a

```C
    addr_ptr = &endpoint->addr;
    if (!(((endpoint->family == AF_INET) && (IS_ACTIVE(SOCK_HAS_IPV4))
            || ((endpoint->family == AF_INET6) && (IS_ACTIVE(SOCK_HAS_IPV46)))) {
        return -ENOTSUP;
    }
```

be easier?

Reply:
Not sure if it's easier on the eyes too :smile: